packages/@expo/cli/src/utils/getOrPromptApplicationId.ts

```diff
@@ -16,14 +18,28 @@ import {
   validatePackage,
   validatePackageWithWarning,
 } from './validateApplicationId';
+import { AppQuery } from '../api/graphql/queries/AppQuery';
 import * as Log from '../log';
-import { getAccountUsername } from './getAccountUsername';
 
 const debug = require('debug')('expo:app-id') as typeof console.log;
 
-function getExpoUsername(exp: ExpoConfig) {
-  // Account for if the environment variable was an empty string.
-  return getAccountUsername(exp) || 'anonymous';
+async function getRecommendedReverseDomainNameSecondPartAsync(
+  exp: ExpoConfig
+): Promise<string | null> {
+  const easProjectId = exp.extra?.eas?.projectId;
```

Comment:
It'll rarely be the case where a user builds their app for the first time after setting up EAS.

Reply:
You mean we expect the more common ordering to be build and then set up EAS? In that case the owner field also wouldn't be set yet since it's `eas init` that sets it (if it's not set manually).